jbpm-runtime-manager/src/main/java/org/jbpm/runtime/manager/impl/migration/MigrationManager.java

```diff
@@ -378,6 +378,9 @@ private void upgradeProcessInstance(KieRuntime oldkruntime,
     private void updateNodeInstances(NodeInstanceContainer nodeInstanceContainer, Map<String, String> nodeMapping, NodeContainer nodeContainer, EntityManager em) {
 
         for (NodeInstance nodeInstance : nodeInstanceContainer.getNodeInstances()) {
+            if (nodeInstance.getNode() == null) {
+                continue;
+            }
```

Comment:
Just a question, a nodeInstance without a node? Is this for ad-hoc nodes?

Reply:
it's a dynamic node - one that does not have any definition